### DISCUSSION TITLE
NumericUpDown Maximum property

### DISCUSSION THREAD

**Discussion by alexachso** [MahApps/MahApps.Metro#4431] [Q&A] [answered]
Hi I would like to set the maximum to a numericUpDown control as fallows:
`Maximum="{x:Static sys:UInt32.MaxValue}"`

But when I run the program I get this error :
![image](https://github.com/MahApps/MahApps.Metro/assets/12586000/54ee3518-e91e-40ca-a849-34fd2f1de258)

Does someone know what am I doing wrong? Thank you!

**Comment by timunie** [comment]
UInt is not supported for NumericUpDown as this Maximum is out of the range we can reach iirc. Check docs for numeric types in .NET

**Reply by alexachso** [reply]
HI, ok :) but I can write `Maximum="4294967295"` and it works just fine. I will check the docs for numeric types, thank you!

**Comment by punker76** [comment, marked as answer]
@alexachso This is a little bit tricky caused by Xaml, but you can do this:

```xaml
<mah:NumericUpDown Maximum="{Binding Source={x:Static sys:UInt32.MaxValue}, Mode=OneWay}"  
                   mah:TextBoxHelper.ClearTextButton="True"
                   Value="10" />
```

![2024-04-16_12h48_36](https://github.com/MahApps/MahApps.Metro/assets/658431/253047be-c267-497d-81cb-7367ea4dfc01)

**Reply by alexachso** [reply]
Thank you!
